Lib/_pyrepl/utils.py

```diff
@@ -63,6 +63,14 @@ class ColorSpan(NamedTuple):
 def str_width(c: str) -> int:
     if ord(c) < 128:
         return 1
+    # gh-139246 for zero-width joiner and combining characters
+    category = unicodedata.category(c)
+    if unicodedata.combining(c):
+        return 0
```

Comment:
```suggestion
    if unicodedata.combining(c):
        return 0
    category = unicodedata.category(c)
```
get category closer to where it's used